Tests/iaas/mandatory-services/mandatory-iaas-services.py

```diff
@@ -179,20 +176,22 @@ def check_for_s3_and_swift(conn: openstack.connection.Connection, s3_credentials
     s3_from_env(s3_creds, "AK", "S3_ACCESS_KEY_ID")
     s3_from_env(s3_creds, "SK", "S3_SECRET_ACCESS_KEY")
 
+    # This is to be used for local debugging purposes ONLY
+    # logger.info(f"using credentials {s3_creds}")
+
     s3 = s3_conn(s3_creds, conn)
-    s3_buckets = list_s3_buckets(s3)
+    s3_buckets = list_s3_buckets(s3) or create_bucket(s3, TESTCONTNAME)
     if not s3_buckets:
-        s3_buckets = create_bucket(s3, TESTCONTNAME)
-        assert s3_buckets
+        raise RuntimeError("failed to create S3 bucket")
 
     # If we got till here, s3 is working, now swift
     swift_containers = list_containers(conn)
     # if not swift_containers:
     #    swift_containers = create_container(conn, TESTCONTNAME)
     result = 0
     if Counter(s3_buckets) != Counter(swift_containers):
-        logger.warning("S3 buckets and Swift Containers differ:\n"
-                       f"S3: {sorted(s3_buckets)}\nSW: {sorted(swift_containers)}")
+        logger.error("S3 buckets and Swift Containers differ:\n"
```

Comment:
Thanks for catching this. As we return 1, indicating an ERROR, we should also log an ERROR. Thanks for catching!